be1-go/message/query_test.go

```diff
@@ -0,0 +1,155 @@
+package message
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/xerrors"
+)
+
+var channel = "/root/channel_id"
+
+func createMessage(i int, timestamp Timestamp) (Message, error) {
+	data, err := NewCreateLAOData("test", timestamp, []byte{byte(i)}, []PublicKey{})
+	if err != nil {
+		return Message{}, err
+	}
+
+	msg := Message{
+		MessageID:         []byte{byte(i)},
+		Data:              data,
+		Sender:            []byte{byte(i)},
+		Signature:         []byte{byte(i)},
+		WitnessSignatures: []PublicKeySignaturePair{},
+	}
+
+	return msg, nil
+}
+
+func compareQueries(q1, q2 Query) (bool, error) {
```

Comment:
I reckon a `(bool, error)` return time can be replaced by an `error` return type where a nil error implies true. You can even have a package level error variable for messages not being equal if you wish to check for that error at some other place

Reply:
I remaned the method and passed the `testing.T` object. I hope it is better now